File: ulyp-ui/src/main/java/com/ulyp/ui/FileRecordingsTabName.java
package com.ulyp.ui;

import com.ulyp.transport.ProcessInfo;
import lombok.Value;

import java.io.File;

@Value
public class FileRecordingsTabName {

    File file;
    ProcessInfo processInfo;

    public FileRecordingsTabName(File file, ProcessInfo processInfo) {
        this.file = file;
        this.processInfo = processInfo;
    }

    @Override
    public String toString() {
        return String.format("%s: %s", file.getAbsolutePath(), processInfo.getMainClassName());
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/FileRecordingsTabName.kt
package com.ulyp.ui

import com.ulyp.core.ProcessMetadata
import java.io.File

class FileRecordingsTabName(private val file: File, private val processMetadata: ProcessMetadata) {
    override fun toString(): String {
        return String.format("%s: %s", file.absolutePath, processMetadata.mainClassName)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileRecordingsTabName

        if (file != other.file) return false
        if (processMetadata != other.processMetadata) return false

        return true
    }

    override fun hashCode(): Int {
        var result = file.hashCode()
        result = 31 * result + processMetadata.hashCode()
        return result
    }


}